### PR TITLE
HWKAPM-685 Remove details from Node and record everything as properti…

### PR DESCRIPTION
**Pull request by objectiser**
…es. Removed the redundant actions from the business txn config and UI.

### CODE REVIEW

**Comment by objectiser** [comment]
@pavolloffay Could you review this please? Cannot be merged until after #647 and will require rebasing (with some changes to align).

Reviewer notes:
* This PR is to remove the 'details' concept on Node and derived events, and instead have all information stored as properties
* As part of doing this work, I removed the SetFault and SetDetail actions in the instrumentation and their equivalent in the business txn config (and UI).
* This also affected the binary annotation mapiing for zipkin spans - which now just store all binary annotations as properties, unless a mapping is defined which explicitly ignores the annotation.

**Comment by objectiser** [comment]
@pavolloffay This PR has been updated to remove conflicts, so is ready for review.

File: api/src/main/java/org/hawkular/apm/api/model/trace/InteractionNode.java [hunk 1]
```diff
@@ -94,9 +98,10 @@ public void setOut(Message out) {
      * @return Whether interaction relates to multiple consumers
      */
     public boolean multipleConsumers() {
-        return getDetails().containsKey(Producer.DETAILS_PUBLISH)
-                && getDetails().get(Producer.DETAILS_PUBLISH)
-                .equalsIgnoreCase(Boolean.TRUE.toString());
+        // TODO: When HWKAPM-698 implemented, it may no longer be necessary to capture this info
+        Set<Property> props=getProperties(Producer.PROPERTY_PUBLISH);
+        return !props.isEmpty()
+                && props.iterator().next().getValue().equalsIgnoreCase(Boolean.TRUE.toString());
```
Comment (pavolloffay) [inline api/src/main/java/org/hawkular/apm/api/model/trace/InteractionNode.java]: What if there are multiple `PROPERTY_PUBLISH`?

**Comment by pavolloffay** [comment]
Code compiles ok, but there are exceptions at runtime. Probably deserialization issues with `zipkin-binary-annotation-mapping.json` as nodeDetails were removed.

**Comment by pavolloffay** [comment]
Could you please fix that? I will run it again after it.

**Comment by objectiser** [comment]
@pavolloffay Thanks for picking that up. Change applied.

**Comment by pavolloffay** [comment]
Now we have kind of strange behavior, properties are `includable` so if there is a mapping for a given key it will be included only if there is `property: {"include": true}` in the mappings, however if there is no mapping for given key it is be included by default. 

Wouldn't be better to change properties to excludable? So if there is no mapping it will be included, if there is mapping it has to be marked as `property: "exclude": true`.

What do you think?

**Comment by objectiser** [comment]
:+1:  yes that would be clearer. I'll update now.

**Comment by objectiser** [comment]
@pavolloffay Could you please re-review. Thanks!

**Comment by objectiser** [comment]
@pavolloffay Ok thanks - could you create a jira with the details of how to reproduce.

**Comment by pavolloffay** [comment]
Created https://issues.jboss.org/browse/HWKAPM-714

**Pull request merged**